scripts/add-erc20-tokens-coingecko.ts

```diff
@@ -0,0 +1,211 @@
+        vs_currency: 'usd',
+        category: categoryId,
+        order: 'market_cap_desc',
+        per_page: numberOfResults,
+        page: 1,
+        sparkline: false,
+        locale: 'en',
+      },
+    },
+  )
+  if (coingeckoResponse.status !== 200 || !coingeckoResponse.data) {
+    throw new Error(`Encountered error fetching tokens list from Coingecko`)
+  }
+
+  const newTokensInfo = [...existingTokensInfo]
+  const fetchFailedTokenIds = []
+  for (let i = 0; i < coingeckoResponse.data.length; i++) {
+    const token = coingeckoResponse.data[i]
+    const { id, image } = token
+    if (!id) {
+      console.warn(`⚠️ No id found for token ${token}`)
+      continue
+    }
+
+    console.log(
+      `(${i + 1}/${coingeckoResponse.data.length}) Processing token ${id}...`,
+    )
+
+    if (existingLowerCaseTokenSymbols.has(token.symbol.toLowerCase())) {
+      console.log(`Token ${id} already exists`)
+      continue
+    }
+
+    // avoid rate limit 10-30 requests / minute
+    // https://apiguide.coingecko.com/getting-started/error-and-rate-limit#rate-limit
+    await new Promise((resolve) => setTimeout(resolve, 10000))
+
+    if (!image) {
+      console.warn(`⚠️ No id or image found for token ${token}`)
+      fetchFailedTokenIds.push(id)
+      continue
+    }
+
+    // get token address from coingecko /coins/{id} endpoint, annoyingly this is
+    // not returned in the /coins/markets response
+    let address = undefined
+    let decimals = undefined
+    try {
+      console.log('Fetching token details from CoinGecko...')
+      const coinDetailResponse = await axios.get(
+        `https://api.coingecko.com/api/v3/coins/${id}`,
+        {
+          params: {
+            localization: false,
+            tickers: false,
+            market_data: false,
+            community_data: false,
+            developer_data: false,
+            sparkline: false,
+          },
+        },
+      )
+      decimals =
+        coinDetailResponse.data.detail_platforms[platformId]?.decimal_place
+      address =
+        coinDetailResponse.data.detail_platforms[platformId]?.contract_address
+      if (!address) {
+        throw new Error(`No token address found for token ${id}`)
+      }
+    } catch (error) {
+      console.warn(
+        `⚠️ Encountered error fetching token address for ${id} from Coingecko: ${error}`,
+      )
+      fetchFailedTokenIds.push(id)
+      continue
+    }
+
+    // get token metadata from token contract directly. Coingecko manually adds
+    // some info to their token list and the token symbol in particular is
+    // always (incorrectly) lowercased.
+    console.log(`Fetching token details from the contract ${address}...`)
+    const [symbol, name] = await Promise.all([
+      client.readContract({
+        address,
+        abi: erc20Abi,
+        functionName: 'symbol',
+      }),
+      client.readContract({
+        address,
+        abi: erc20Abi,
+        functionName: 'name',
+      }),
+    ])
```

Comment:
nit: doesn't really matter here, but just for the sake of getting in the habit of minimizing RPC calls
```suggestion
    const [symbol, name] = await client.multicall({
      contracts: [
        {
          address,
          abi: erc20Abi,
          functionName: 'symbol',
        },
        {
          address,
          abi: erc20Abi,
          functionName: 'name',
        }
      ],
      allowFailure: false,
    })
```